arangod/Cluster/AutoRebalance.cpp

```diff
@@ -230,29 +230,36 @@ ShardImbalance AutoRebalanceProblem::computeShardImbalance() const {
 
 std::vector<double> AutoRebalanceProblem::piCoefficients(
     Collection const& c) const {
-  std::vector<double> tmp;
-  tmp.resize(dbServers.size(), 0);
+  std::vector<double> leaders;
+  leaders.resize(dbServers.size(), 0);
+  std::vector<bool> haveShards;
+  haveShards.resize(dbServers.size(), 0);
+  uint32_t dbServersAffected = 0;
   if (c.shards.size() == 1) {
```

Comment:
This if condition can now be moved up before the initialization of `haveShards` and `dbServersAffected`, because the latter have no influence on the if condition and what's happening in the if condition's execution block.